### PR TITLE
Milestone-4: Filter books in the table by category.

### PR DESCRIPTION
**Pull request by LamiaSristy**
# Here we have added filtering for the category book. We have added the following files to do so:

1. Added a new action `CHANGE_FILTER` in `actions/index.js`.
2. Created a new `filter.js` in the reducers.
3. Added a new component `CategoryFilter`.

# Now a user can select a category from the dropdown and choose the required category.

### CODE REVIEW

File: src/components/CategoryFilter.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import { connect } from 'react-redux';
+import { catFilter } from '../actions/index';
+
+const mapDispatchToProps = dispatch => ({
+  catFilter: filter => {
+    dispatch(catFilter(filter));
+  },
+});
+
+const CategoryFilter = ({ catFilter }) => {
```
Comment (addod19) [inline src/components/CategoryFilter.js:5]: Kindly change this component to a presentational component

File: src/containers/BookList.js [hunk 2]
```diff
@@ -2,41 +2,50 @@ import React from 'react';
...
-        <tr>
-          <th>ID</th>
-          <th>title </th>
-          <th>category</th>
-        </tr>
-      </thead>
-      <tbody>{books.map(renderBook)}</tbody>
-    </table>
+    <div>
+      <CategoryFilter />
```
Comment (addod19) [inline src/containers/BookList.js]: Kindly pass the handleFilterChange down to the categoryFilter

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi, @LamiaSristy and @JelilFaisalAbudu, great job on this project :+1: PR is descriptive, it shows what work you did in this milestone :100: Keep up :+1: The README file is also professional :1st_place_medal: And the project works as expected with the filter action being implemented. However, there are a few issues that need addressing. Kindly check the comments below:
- `BookList component` - It implements handleFilterChange and pass it down to the categoryFilter. Currently, the handleFilterChange is not getting passed down to the category filter in the booklist component as in the screenshot below:
bf
- `CategoryFilter` component must be a presentational component. Presentational components are concerned with how things look but not how things work which we have currently with dispatching an action to update the store. Let us kindly fix this

[Optional]
- Do we really need a "none" category? I created a data with the "none" category but the category table was empty which makes the app not so cool
- Also, let us add a simple validation so we cannot create an empty data
- It would be great if we can update our live link 


Once these are fixed, kindly submit another code review :smile: 
_Happy coding_ :woman_technologist: :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

File: src/components/CategoryFilter.js (hunk 1 above)
Comment (LamiaSristy) [reply src/components/CategoryFilter.js:5]: Done

File: src/containers/BookList.js (hunk 2 above)
Comment (LamiaSristy) [reply src/containers/BookList.js]: Done.

**Review by shloch — APPROVED** [review]
`CategoryFilter.js` has been turned to presentational component
and the right prop has been passed to the `Booklist` component.

Based on these recommendations of the former TSEs, you get a thumbs up.

accomplished

Happy coding

**Pull request merged**